Kudu.Core/Permissions/PermissionHandler.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.IO;
+using Kudu.Contracts.Permissions;
+using Kudu.Contracts.Settings;
+using Kudu.Core.Deployment;
+using Kudu.Core.Deployment.Generator;
+using Kudu.Core.Infrastructure;
+
+namespace Kudu.Core.Permissions
+{
+    /// <summary>
+    /// This handler is meant for Linux enviroment only
```

Comment:
Since the context should never affect what it does, maybe you can have a singleton instance that gets created early in Ninject init (not request context), and then is used everywhere? Well, not sure if that would mess up the logger. Maybe discuss with @suwatch. But it does seem a bit overkill for such a simple need.